src/vm/docs/mod.rs

```diff
@@ -595,6 +595,20 @@ If the function returns _ok_, database changes occurred.",
     example: "(contract-call? .tokens transfer 'SZ2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQ9H6DPR 19) ;; Returns (ok 1)"
 };
 
+const TRAIT_PRINCIPAL_API: SpecialAPI = SpecialAPI {
+    input_type: "Trait",
+    output_type: "principal",
+    signature: "(trait-principal .contract-name)",
+    description: "The `trait-principal` function returns the contract implementing the trait.
+The contract must explicitly use `impl-trait` for the call to be valid.",
```

Comment:
Looking at your latest changes, I think we should update this, it looks like we do support "accidental" traits implementations.

Reply:
ah, yes.  Good catch, now that the implementation has changed, the doc should reflect this